base/services/w32time/w32time.c

```diff
@@ -282,26 +286,53 @@ W32TmServiceMain(DWORD argc, LPWSTR *argv)
     /* The worker loop of a service */
     for (;;)
     {
-        result = SetTime();
+        if (RegOpenKeyExW(HKEY_LOCAL_MACHINE,
```

Comment:
Do I read the code right, that it will read the registry in an endless loop each second? If yes, that should be avoided to improve performance.

Reply:
The default interval is 7 days:

![VirtualBox_ReactOS_02_11_2023_20_16_45](https://github.com/reactos/reactos/assets/2107452/af585c61-cde3-42ca-8e68-b3fc46e51852)

Reply:
The question is what really happens if this value gets deleted for whatever reason.

Reply:
Thanks for explaining what the dwInterval is set to. My concern can be considered as being addressed. I will keep the thread open until hbelusca will also confirm that his concern has been addressed.
So please wait with that until the merging, although I will approve already. Thx.

Reply:
Would it be possible to listen for a message when the key changes?
(Do we know how Windows handles this key?)

Reply:
It quite possibly uses registry notifications, see https://learn.microsoft.com/en-us/windows/win32/api/winreg/nf-winreg-regnotifychangekeyvalue
However we don't implement that functionality yet. Hence we can only poll, for the time being.

Reply:
> we don't implement that functionality yet. Hence we can only poll

Ah. Then, add a "TODO: Use RegNotifyChangeKeyValue() when implemented" comment?
https://github.com/reactos/reactos/blob/c6c6c62cb5b44468aad38b0604339a97f80b1b69/dll/win32/advapi32/reg/reg.c#L3169

Reply:
> Ah. Then, add a "TODO: Use RegNotifyChangeKeyValue() when implemented" comment?

Yes that would be a suggestion for the present PR.

Reply:
Comment added.